File: code/py/linear-search.py
def linear_search(list:list, target):
    """
    Returns the index position of the target value from the given list
    """
    for i in range(0, len(list)):
        if list[i] == target:
            return i
    return None

input = range(0, 1001)
result = linear_search(input, 1000)
assert result == 1000

input = range(0, 1001)
result = linear_search(input, 2000)
assert result is None

File: code/julia/linear-search.jl

using(Test)

"""
Returns the index position of the target value from the given list
"""
function linear_search(input::Vector, target::Int)::Union{Int, Nothing}
    for i in input
        if(i == target) return i end
    end
end

input = collect(range(1, 500))
result = linear_search(input, 500)
@test result == 500

input = collect(range(1, 1000))
result = linear_search(input, 1001)
@test isnothing(result)